crates/turborepo-lib/src/run/mod.rs

```diff
@@ -229,10 +230,26 @@ impl Run {
             self.base.ui,
         );
 
+        let mut global_env_mode = opts.run_opts.env_mode;
+        if matches!(global_env_mode, EnvMode::Infer)
+            && !root_turbo_json.global_pass_through_env.is_empty()
+        {
+            global_env_mode = EnvMode::Strict;
+        }
+
         let pkg_dep_graph = Arc::new(pkg_dep_graph);
         let engine = Arc::new(engine);
-        let visitor = Visitor::new(pkg_dep_graph, &opts);
-        visitor.visit(engine).await?;
+        let visitor = Visitor::new(
```

Comment:
Is there a reason why we're no longer constructing the task hash tracker and calculating file hashes before we start walking the graph?